godtools/App/Features/Tutorial/TutorialFlow.swift

```diff
@@ -24,6 +26,7 @@ class TutorialFlow: Flow {
         
         self.flowDelegate = flowDelegate
         self.appDiContainer = appDiContainer
+        self.flowDiContainer = TutorialFlowDiContainer(appDiContainer: appDiContainer)
```

Comment:
Hey @reldredge71 basically I'm testing a slightly different structure using clean architecture (adding domain layer) with this specific feature with the possibility of moving our other features to this format.  
The main idea is to
1. Move specific feature dependencies into each feature folder and use the flowDiContainer to fetch any useCases or viewModel dependencies specific to that feature.  This will also help reduce clutter in the AppDiContainer and make it easier to see which features depend on what.  AppDiContainer would still contain any shared or core dependencies that are used across features.
2. Move business logic out of the viewModel and into useCases to help with re-using useCases across viewModels and to reduce viewModel responsibilities.

I'm glad you noticed that and asked.  Also, having a feature specific dependency container (TutorialFlowDiContainer) allows for the possibility of injecting containers into the feature that may point to different data layers and useCase implementations.  In our case we're instantiating TutorialFlowDiContainer directly, but it opens up this possibility.

Reply:
I'll try to find an example of where I've seen this setup used.  I believe GodTools Android uses something similar. @gyasistory  would know.